app/server/models/credit_transfer.py

```diff
@@ -172,19 +174,67 @@ def resolve_as_completed(self, existing_blockchain_txn=None):
             if self.recipient_user and self.recipient_user.transfer_card:
                 self.recipient_user.transfer_card.update_transfer_card()
 
-        if self.transfer_type == TransferTypeEnum.DEPOSIT and self.fiat_ramp:
+        if self.fiat_ramp and self.transfer_type in [TransferTypeEnum.DEPOSIT, TransferTypeEnum.WITHDRAWAL]:
             self.fiat_ramp.resolve_as_completed()
```

Comment:
I just realised: the exchange has the order of control the other way round - resolving an exchange as complete resolves the transfers as complete. (both are equivalently valid based on the context AFAIK). What's the process that triggers this in the fiat ramp flow?

Reply:
In the PoLI payments API you call `deposit. resolve_as_completed()` to resolve the `make_deposit_transfer()` utils fn...